pkg/telemetry/telemetryservice.go

```diff
@@ -56,15 +56,20 @@ func NewTelemetryService(notifier webhook.Notifier, analytics AnalyticsService)
 }
 
 func (t *telemetryService) run() {
+	lastUpdatedAt := time.Now()
 	ticker := time.NewTicker(updateFrequency)
 	for {
 		select {
 		case <-ticker.C:
-			t.internalService.SendAnalytics()
+			lastUpdatedAt = t.internalService.SendAnalytics()
 		case job, ok := <-t.jobQueue:
 			if ok {
 				job()
 			}
+			// check if we missed sending analytics every updateFrequency seconds
+			if time.Now().After(lastUpdatedAt.Add(updateFrequency)) {
+				lastUpdatedAt = t.internalService.SendAnalytics()
+			}
```

Comment:
Any idea why this would happen? Too many events overwhelming the job queue?

Reply:
@boks1971 I suspected stats in queue leading to missed ticker events. Don't think that is the issue. Will add debug logs to capture whether the delta stats was leading to missed intervals